hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/impl/BurnPrecompile.java

```diff
@@ -145,6 +145,10 @@ public static BurnWrapper getBurnWrapper(final Bytes input, @NonNull final Syste
         final var fungibleAmount = SystemContractAbis.toLongSafely(decodedArguments.get(1));
         final var serialNumbers = (long[]) decodedArguments.get(2);
 
+        if (fungibleAmount < 0 || (fungibleAmount == 0 && serialNumbers.length == 0)) {
```

Comment:
Do we need to throw for `(fungibleAmount == 0 && serialNumbers.length == 0)` ?

Reply:
It's not critical for this PR, but in general I would suggest only throwing an exception inside `Precompile.body()` when the input is unparseable.

If the input is parseable but invalid, we can surface a more meaningful response code to the user by throwing an `InvalidTransactionException` inside `precompile.run()`.

Reply:
E.g. if we just parsed this into `burnOP`, then we could begin,
```
    @Override
    public void run(final MessageFrame frame) {
        Objects.requireNonNull(burnOp, "`body` method should be called before `run`");
        validateTrue(burnOp.amount() >= 0, INVALID_TOKEN_BURN_AMOUNT);
```

Reply:
Good suggestions @tinker-michaelj .  Thanks makes sense.